sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,729 @@
+
+This extension also depends on the following other SYCL extensions:
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties]
+* link:../experimental/sycl_ext_oneapi_enqueue_functions.asciidoc[
+  sycl_ext_oneapi_enqueue_functions]
+* link:../supported/sycl_ext_oneapi_default_context.asciidoc[
+  sycl_ext_oneapi_default_context]
+
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Backend support status
+
+This extension is not yet implemented in {dpcpp}.
+
+== Overview
+
+This extension introduces an interface for allocating and freeing USM memory
+asynchronously, as well as introducing a USM memory pool from which this memory
+is allocated and freed from.
+
+The asynchornous memory allocation and free are exposed via new free functions
+taking a queue, which enqueue an asynchronous malloc or free command and return
+immediately. The functions can take a memory pool object in order to allocate
+from that memory pool specifically, however, if no memory pool object is
+provided there is a default memory pool which will be used.
+
+The memory pool introduced is a dynamic memory pool, as such no memory is
+immediately allocated on construction, instead memory is allocated by the OS
+for the pool when allocations are made and are freed back to the OS when the
+queue which enqueued the malloc and free commands is synchronized with.
+
+The immediate benefit of using the asynchornous malloc and free functions, as
+opposed to the existing synchronous malloc and free functions, is that they no
+longer block the calling thread. So applications which wish to allocate
+temporary USM memory but not carry around the pointer no longer have to
+synchronize with the execution of kernels accessing that memory and call the
+free function before continuing.
+
+A further benefit comes from the use of a memory pool for allocations, as
+malloc and free commands are asynchronous, the runtime can opportunistically
+re-use memory allocations from one kernel to another, without any intermediate
+freeing and re-allocation. This benefit can be further extended beyond queue
+synchronization by specify a threshold, which will instruct the runtime to (if
+possible) maintain a certain size of memory and not release that back to the OS
+even when the queue is synchronized with.
+
+There are also other properties which can be used when constructing a memory
+pool object to control other aspects of how the memory is allocated.
+
+== Integration with other extensions
+
+It is a priority for this extension to integrate with the
+link:../supported/sycl_ext_oneapi_graph.asciidoc[
```

Comment:
```suggestion
link:../experimental/sycl_ext_oneapi_graph.asciidoc[
```

Reply:
Fixed in https://github.com/intel/llvm/pull/14800/commits/2c51adf28ece21c0e91c2c0ec85440ae1153e536.